### PR TITLE
feat(route): add xueqiu timeline

### LINKED ISSUE #4564: 雪球用户个人时间线的RSS
### 这是一个什么样的功能？
雪球用户个人时间线的RSS

### 这个功能可以解决什么问题？
雪球用户有个人时间线, 汇总了所有关注的人的更新 (包括讨论). 如果用RSSHUB的单人RSS比较杂乱,维护更新差. 希望有个RSS直接接驳个人时间线. 

### 额外描述
例子: https://xueqiu.com/v4/statuses/home_timeline.json?source=user&_=1587857022238
image

### PR DESCRIPTION
**Pull request by ErnestDong**
## Involved Issue / 该 PR 相关 Issue

Close #4564

## Example for the Proposed Route(s) / 路由地址示例

```routes
/xueqiu/timeline
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [x] Anti-bot or rate limit / 反爬/频率限制
  - [x] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

增加雪球关注时间线，由于需要雪球cookie只能自建。相较于单独添加每个用户而言能同步所有已关注的用户动态，雪球时间线如图：

image

效果如图：
image

Demo链接：[链接](http://18.189.188.195:1200/xueqiu/timeline?key=ernest&limit=50)

查询参数`usergroup_id`为分组id，-1是全部，-2是关注精选，用户自己的分组是1、2、3……

### CODE REVIEW

File: lib/routes/xueqiu/timeline.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+};
+
+async function handler(ctx) {
+    const cookie = config.xueqiu.cookies;
+    const limit = ctx.req.query('limit') || 15;
+    const usergroup_id = ctx.req.query('usergroup_id') || 1;
+    if (cookie === undefined) {
+        throw new Error('缺少雪球用户登录后的 Cookie 值');
+    }
+    let out = [] as any[];
```
Comment (TonyRL) [inline lib/routes/xueqiu/timeline.ts:40]: Do not use `any`
https://github.com/DIYgod/RSSHub/blob/d748a8c92e714a5bea9526ed9abf8191c487ccab/lib/types.ts#L30

File: lib/routes/xueqiu/timeline.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    let max_id = -1;
+
+    async function fetchItems() {
+        const data = await fetchNextID(max_id, cookie as string, usergroup_id);
+        const items = await Promise.all(
+            data.home_timeline.map((item) =>
+                cache.tryGet(item.target, async () => {
+                    const retweetedStatus = (await item.retweeted_status) ? `<blockquote>${item.retweeted_status.user.screen_name}:&nbsp;${item.retweeted_status.description}</blockquote>` : '';
```
Comment (TonyRL) [inline lib/routes/xueqiu/timeline.ts:48]: `item.retweeted_status` does not require `await`

File: lib/routes/xueqiu/timeline.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    name: '用户关注时间线',
+    maintainers: ['ErnestDong'],
+    handler,
+    description: `:::warning
+  用户关注动态需要登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。
+  :::`,
```
Comment (TonyRL) [inline lib/routes/xueqiu/timeline.ts:30]: Please do mention what `usergroup_id` means in the docs

Comment (ErnestDong) [reply lib/routes/xueqiu/timeline.ts]: Changed `usergroup_id` to `param` instead of `query` and added docs. I'll push to demo later.

**Comment by ErnestDong** [comment]
updated with [demo link](http://18.189.188.195:1200/xueqiu/timeline?key=ernest)

File: lib/routes/xueqiu/timeline.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    name: '用户关注时间线',
+    maintainers: ['ErnestDong'],
+    handler,
+    description: `:::warning
+  用户关注动态需要登录后的 Cookie 值，所以只能自建，详情见部署页面的配置模块。
+  :::
+
+  |-1  |-2      | 1         |
+  |----|--------|-----------|
+  |全部|关注精选|自定义第1组|`,
```
Comment (TonyRL) [inline lib/routes/xueqiu/timeline.ts:34]:
```suggestion
  | -1   | -2       | 1             |
  | ---- | -------- | ------------- |
  | 全部 | 关注精选 | 自定义第 1 组 |`,
```

**Pull request merged**